dynamiqs/mesolve/mesolve.py

```diff
@@ -85,12 +84,33 @@ def mesolve(
             - **gradient** _(Gradient)_ -- Gradient used.
             - **options** _(Options)_ -- Options used.
     """
+    # === convert arguments
+    H = _astimearray(H)
+    jump_ops = [_astimearray(jump_op) for jump_op in jump_ops]
+    rho0 = jnp.asarray(rho0, dtype=cdtype())
+    rho0 = todm(rho0)
+    tsave = jnp.asarray(tsave)
+    exp_ops = jnp.asarray(exp_ops, dtype=cdtype()) if exp_ops is not None else None
+
+    return _vmap_mesolve(H, jump_ops, rho0, tsave, exp_ops, solver, gradient, options)
+
+
+@partial(jax.jit, static_argnames=('solver', 'gradient', 'options'))
+def _vmap_mesolve(
+    H: TimeArray,
+    jump_ops: list[TimeArray],
+    rho0: ArrayLike,
+    tsave: ArrayLike,
+    exp_ops: list[ArrayLike] | None = None,
+    solver: Solver = Tsit5(),  # noqa: B008
+    gradient: Gradient | None = None,
+    options: Options = Options(),  # noqa: B008
+) -> Result:
     # === vectorize function
```

Comment:
Why do we need to declare an interemediary function here? Couldn't we merge `mesolve` and `_vmap_mesolve`?

Reply:
This is to handle QObjs as inputs: we intentionnally delay the `jit` so there is no tracing on QObjs inputs. See also the comment of @pierreguilmin above.